src/js/plugins/plugin.url.js

```diff
@@ -154,12 +154,27 @@ BookReader.prototype.urlUpdateFragment = function() {
       window.history.replaceState({}, null, newUrlPath);
     }
   } else {
-    window.location.replace('#' + newFragment + newQueryString);
+    const newQueryStringSearch = this.urlParamsFiltersOnlySearch(this.readQueryString());
+    window.location.replace('#' + newFragment + newQueryStringSearch);
   }
 
   this.oldLocationHash = newFragment;
 };
 
+/**
+ * filtering query parameters to select only book search param (?q=foo)
+   this needs to be updated/URL system modified if future query params are to be added
+ * @param {string} url
+ * @return {string}
+ * */
+BookReader.prototype.urlParamsFiltersOnlySearch = function(url) {
+  const queries = new URLSearchParams(url);
+  const searchedTerm = queries.get('q');
+  const newQueryStringSearch = searchedTerm == null ? "" : 'q=' + searchedTerm;
+  return newQueryStringSearch;
```

Comment:
This might break if the query has special characters in it (like `&`), since `.get(...)` returns the _unencoded_ parameters. Add a unit test for it; using `URLSearchParams#toString` should make sure we're good :+1:

```suggestion
  const params = new URLSearchParams(url);
  return params.has('q') ? new URLSearchParams({ q: params.get('q') }).toString() : '';
```

Reply:
DONE